### PR TITLE
PLT-1454: Added user management support 

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: client/user.go [hunk 1]
```diff
@@ -56,6 +57,32 @@ func (h *V1Client) GetUsers() (*models.V1Users, error) {
...
+	summaryParams := clientv1.NewV1UsersSummaryGetParams().WithBody(filterSummary)
+	summaryResponse, err := h.Client.V1UsersSummaryGet(summaryParams)
+	if err != nil {
+		return nil, err
+	}
+	if summaryResponse.Payload.Items != nil {
+		if len(summaryResponse.Payload.Items) == 1 {
+			return summaryResponse.Payload.Items[0], nil
+		}
+		return nil, errors.New("More than one user found name: " + userEmail)
```
Comment (TylerGillson) [inline client/user.go:80]: "More than one user found with email:"

File: client/user.go [hunk 2]
```diff
@@ -117,3 +154,103 @@ func (h *V1Client) CreateUser(user *models.V1UserEntity) (string, error) {
 	}
 	return *resp.Payload.UID, nil
 }
+
+// UpdateUser update existing user.
+func (h *V1Client) UpdateUser(uid string, user *models.V1UserUpdateEntity) error {
+	param := clientv1.NewV1UsersUIDUpdateParams().WithUID(uid).WithBody(user)
+	_, err := h.Client.V1UsersUIDUpdate(param)
```
Comment (TylerGillson) [inline client/user.go:161]: just return h.Client.V1UsersUIDUpdate(param)

File: client/user.go [hunk 3]
```diff
@@ -117,3 +154,103 @@ func (h *V1Client) CreateUser(user *models.V1UserEntity) (string, error) {
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// AssociateUserProjectRole associate project role for the user.
+func (h *V1Client) AssociateUserProjectRole(userUID string, body *models.V1ProjectRolesPatch) error {
+	param := clientv1.NewV1UsersProjectRolesPutParams().WithUID(userUID).WithBody(body)
+	_, err := h.Client.V1UsersProjectRolesPut(param)
```
Comment (TylerGillson) [inline client/user.go:171]: return h.Client.V1UsersProjectRolesPut(param)

File: client/user.go [hunk 4]
```diff
@@ -117,3 +154,103 @@ func (h *V1Client) CreateUser(user *models.V1UserEntity) (string, error) {
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// AssociateUserTenantRole associate tenant role to the user.
+func (h *V1Client) AssociateUserTenantRole(userUID string, body *models.V1UserRoleUIDs) error {
+	param := clientv1.NewV1UsersUIDRolesUpdateParams().WithUID(userUID).WithBody(body)
+	_, err := h.Client.V1UsersUIDRolesUpdate(param)
```
Comment (TylerGillson) [inline client/user.go:241]: return h.Client.V1UsersUIDRolesUpdate(param)

File: client/user.go (hunk 1 above)
Comment (SivaanandM) [reply client/user.go]: updated

File: client/user.go (hunk 2 above)
Comment (SivaanandM) [reply client/user.go:161]: h.Client.V1UsersUIDUpdate(param) will return 2 arguments (*V1UsersUIDUpdateNoContent, error) hence we are returning only err

Comment (TylerGillson) [reply client/user.go:161]: right, ok. the error check is still not needed. You can just return err.

Comment (SivaanandM) [reply client/user.go:158]: done

File: client/user.go (hunk 3 above)
Comment (SivaanandM) [reply client/user.go:165]: remove err check and return err directly

File: client/user.go [hunk 5]
```diff
@@ -117,3 +154,103 @@ func (h *V1Client) CreateUser(user *models.V1UserEntity) (string, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	return resp.Payload.ResourceRoles, nil
+}
+
+// DeleteUserResourceRoles delete resource role for the user.
+func (h *V1Client) DeleteUserResourceRoles(userUID string, roleUID string) error {
+	param := clientv1.NewV1UsersUIDResourceRolesUIDDeleteParams().WithUID(userUID).WithResourceRoleUID(roleUID)
+	_, err := h.Client.V1UsersUIDResourceRolesUIDDelete(param)
```
[the remark this answers is not in the document]
Comment (SivaanandM) [reply client/user.go:216]: remove err check and return err directly

File: client/user.go (hunk 4 above)
Comment (SivaanandM) [reply client/user.go:223]: remove err check and return err directly

**Review by TylerGillson — APPROVED** [review]

**Pull request merged**